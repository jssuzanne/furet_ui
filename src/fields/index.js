/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import plugin from '../plugin';
import Vue from 'vue';
import './string';
// import './selection';
// import './date';
// import './datetime';
// import './time';
import './integer';
// import './float';
// import './url';
// import './uuid';
// import './sequence';
// import './password';
// import './color';
// import './text';
// import './boolean';
// import './json';
// import './x2one';
// import './x2many';
// import './largebinary';

/**
 * Unknown field, used only if the wanted field is not available
**/
const Field = Vue.component('furet-ui-field-unknown', {
    props: ['config', 'values'],
    template: `
        <div>
            {{ config }}
            {{ values }}
        </div>
    `,
    computed: {
        value () {
            return this.parentProps.row[this.config.name];
        }
    }
});

plugin.set(['field'], {Unknown: 'furet-ui-field-unknown'});
