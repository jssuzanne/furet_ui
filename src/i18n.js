/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
export default {
    locale: 'en',
    fallbackLocale: 'en',
    messages: {
        en: {
            menus: {
                close: 'Close',
                search: 'Search ...',
            },
            views: {
                unknown: {
                    title: 'The wanted view "{name}" is unknown',
                    message: "Report this message to the administrator",
                },
                common: {
                    create: 'Create',
                    save: 'Save',
                    edit: 'Edit',
                    cancel: 'Cancel',
                    delete: 'Delete',
                    close: 'Close',
                    actions: 'Actions',
                    more: 'More',
                },
                clients: {
                    login: {
                        button: 'Log in',
                    },
                },
            },
        },
    },
}
