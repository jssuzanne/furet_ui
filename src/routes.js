/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import Vue from 'vue';
import VueRouter from 'vue-router';
Vue.use(VueRouter);
import {dispatchAll} from './store';
import {json_post} from './server-call';


const spaceChildren = [
    {
        path: 'action/:actionId',
        props: true,
        component: {
            template: '<furet-ui-space v-bind:actionId="actionId" />',
            props: ['actionId'],
        },
        beforeEnter: (to, from, next) => {
            next()
        },
        children: [
            {
                path: 'view/:viewId',
                props: true,
                component: {
                    template: '<furet-ui-space v-bind:viewId="viewId" />',
                    props: ['viewId'],
                },
                beforeEnter: (to, from, next) => {
                    console.log('view', to, from)
                    next()
                },
            },
        ],
    },
];



const router = new VueRouter({
    routes: [
        {
            path: '/custom/view/:viewName',
            name: 'custom_view',
            props: true,
            component: {
                template: '<furet-ui-custom-view v-bind:viewName="viewName" />',
                props: ['viewName'],
                beforeRouteEnter: (to, from, next) => {
                    json_post('/custom/view/' + to.params.viewName, {}, {
                        onSucess: (results) => {
                            dispatchAll(undefined, results);
                        },
                    })
                    next();
                },
                beforeRouteUpdate: (to, from, next) => {
                    json_post('/custom/view/' + to.params.viewName, {}, {
                        onSucess: (results) => {
                            dispatchAll(undefined, results);
                        },
                    })
                    next();
                },
            },
        },
        {
            path: '/space/:spaceId',
            name: 'space',
            props: true,
            component: {
                template: '<furet-ui-space v-bind:spaceId="spaceId" />',
                props: ['spaceId'],
                beforeRouteEnter: (to, from, next) => {
                    call('/space/' + to.params.spaceId, to.params);
                    next();
                },
                beforeRouteUpdate: (to, from, next) => {
                    if (to.params.spaceId != from.params.spaceId) call('/space/' + to.params.spaceId, to.params);
                    next();
                },
            },
            children: spaceChildren,
        },
        {
            path: '/space/:spaceId/menu/:menuId',
            name: 'space_menu',
            props: true,
            component: {
                template: '<furet-ui-space v-bind:spaceId="spaceId" v-bind:menuId="menuId" />',
                props: ['spaceId', 'menuId'],
                beforeRouteEnter: (to, from, next) => {
                    console.log('plop', to.params)
                    call('/space/' + to.params.spaceId, to.params);
                    next();
                },
                beforeRouteUpdate: (to, from, next) => {
                    console.log('plop', to.params)
                    if (to.params.spaceId != from.params.spaceId) call('/space/' + to.params.spaceId, to.params);
                    next();
                },
            },
            children: [
                {
                    path: 'action/:actionId',
                    name: 'space_menu_action',
                    props: true,
                    component: {
                        template: '<div>{{spaceId}} {{menuId}} {{actionId}} </div>',
                        props: ['spaceId', 'menuId', 'actionId'],
                    },
                    beforeEnter: (to, from, next) => {
                        console.log('action', to, from)
                        next()
                    },
                    children: [
                        {
                            path: 'view/:viewId',
                            props: true,
                            component: {
                                template: '<furet-ui-space v-bind:viewId="viewId" />',
                                props: ['viewId'],
                            },
                            beforeEnter: (to, from, next) => {
                                console.log('view', to, from)
                                next()
                            },
                        },
                    ],
                },
            ],
            // children: [
            //     {
            //         path: 'action/:actionId',
            //         name: 'space_menu_action',
            //         props: true,
            //         component: {
            //             template: '<furet-ui-space v-bind:spaceId="spaceId" v-bind:menuId="menuId" v-bind:actionId="actionId"/>',
            //             props: ['spaceId', 'menuId', 'actionId'],
            //             beforeRouteEnter: (to, from, next) => {
            //                 console.log('action.beforeRouteEnter', to)
            //                 next();
            //             },
            //             beforeRouteUpdate: (to, from, next) => {
            //                 console.log('action.beforeRouteUpdate', to)
            //                 next();
            //             },
            //         }
            //     }
            // ],
        },
    ],
});

const call = (path, params) => {
    json_post(path, params, {
        onSuccess: (results) => {
            dispatchAll(router, results);
        },
    })
}

export default router;
