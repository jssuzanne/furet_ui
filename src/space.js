/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import Vue from 'vue';
import {dispatchAll} from './store';
import {json_post} from './server-call';

Vue.component('furet-ui-space-menu', {
    template: `
        <ul class="menu-list">
            <li v-for="menu in menus">
                <span v-if="(menu.submenus || []).length != 0">
                    <furet-ui-picture 
                        v-bind:type="menu.image.type"
                        v-bind:value="menu.image.value"
                    />
                    {{menu.label}}
                </span>
                <furet-ui-space-menu 
                    v-if="(menu.submenus || []).length != 0"
                    v-bind:menus="menu.submenus || []"
                    v-bind:menuId="menuId"
                    v-bind:spaceId="spaceId"
                />
                <a v-if="menu.actionId || menu.custom_view"
                   v-on:click="onClick(menu)"
                   v-bind:class="[menu.id == menuId ? 'is-active' : '']"
                >
                    <furet-ui-picture 
                        v-bind:type="menu.image.type"
                        v-bind:value="menu.image.value"
                    />
                    {{menu.label}}
                </a>
            </li>
        </ul>`,
    props: ['menus', 'menuId', 'spaceId'],
    methods: {
        onClick (menu) {
            if (menu.actionId) {
                this.$router.push({
                    name: 'space_menu_action',
                    params: {
                        spaceId: this.spaceId,
                        menuId: menu.id,
                        actionId: menu.actionId,
                    },
                });
            }
            dispatchAll([
                {type: 'RESET_ACTION_MANAGER'},
                {type: 'CLEAR_ALL_CHANGES'},
            ]);
        }
    }
});

export const Space = Vue.component('furet-ui-space', {
    template: `
        <div class="columns is-gapless">
            <div v-if="isOpenLeft && left_menu.length > 0" class="column is-one-quarter is-half-mobile">
                <aside class="menu">
                    <furet-ui-space-menu 
                        v-bind:menus="left_menu" 
                        v-bind:menuId="menuId" 
                        v-bind:spaceId="spaceId"
                    />
                </aside>
            </div>
            <div class="column">
                <nav class="nav">
                    <div class="nav-left">
                        <a class="button" v-on:click="isOpenLeft = !isOpenLeft" v-if="left_menu.length > 0">
                            <i class="fa fa-bars fa-2x" aria-hidden="true"></i>
                        </a>
                        <ul v-bind:style="{padding: '10px 0px', listStyle: 'none'}">
                            <li v-bind:style="{display: 'inline'}" 
                                v-for="a in actions"
                            >
                                <a v-on:click="onClick(a)">
                                    {{a.label}}
                                </a>
                                /
                            </li>
                            <li v-bind:style="{display: 'inline'}" >
                                {{action.label}}
                            </li>
                        </ul>
                    </div>
                    <div class="nav-right">
                        <div class="field has-addons" v-if="action.views.length > 0">
                            <p class="control" v-for="view in action.views">
                                <b-tooltip v-bind:label="view.type" position="is-left" type="is-info">
                                    <a class="button" 
                                        v-on:click.stop="changeView(view.viewId)"
                                        v-bind:disabled="view.viewId == viewId"
                                        v-bind:class="[view.viewId == viewId ? 'is-primary': '']"
                                    >
                                        <span class="icon is-small">
                                            <furet-ui-view-icon v-bind:type="view.type" />
                                        </span>
                                    </a>
                                </b-tooltip>
                            </p>
                        </div>
                        <a class="button" v-on:click="isOpenRight = !isOpenRight" v-if="right_menu.length > 0">
                            <i class="fa fa-bars fa-2x" aria-hidden="true"></i>
                        </a>
                    </div>
                </nav>
                <router-view></router-view>
            </div>
            <div v-if="isOpenRight && right_menu.length > 0" class="column is-one-quarter is-half-mobile">
                <aside class="menu">
                    <furet-ui-space-menu 
                        v-bind:menus="right_menu" 
                        v-bind:menuId="menuId" 
                        v-bind:spaceId="spaceId"
                    />
                </aside>
            </div>
        </div>`,
    props: ['spaceId', 'menuId', 'actionId', 'viewId'],
    data: () => {
        const data = {
            isOpenLeft: false,
            isOpenRight: false,
        }
        return data
    },
    computed: {
        space_state () {
            return this.$store.state.global.spaces[String(this.spaceId)];
        },
        left_menu () {
            return this.space_state && this.space_state.left_menu || [];
        },
        right_menu () {
            return this.space_state && this.space_state.right_menu || [];
        },
        actions () {
            return this.$store.state.global.breadscrumbs || [];
        },
        action () {
            if (this.actionId) {
                const action = this.$store.state.data.actions[String(this.actionId)];
                if (action) return {label: action.label, views: action.views || []}
           }
           return {label: '', views: []};
        }
    },
    methods: {
        changeView (viewId) {
            this.$store.commit('CLEAR_ALL_CHANGE');
            this.$router.push({
                name: this.menuId ? 'space_menu_action_view' : 'space_action_view',
                params: {
                    spaceId: this.spaceId,
                    menuId: this.menuId,
                    actionId: this.actionId,
                    viewId
                }
            });
        },
        onClick (action) {
            this.$store.commit('REMOVE_FROM_BREADSCRUMB', {position: action.position});
            this.$store.commit('REPLACE_CHANGE', {changes: action.changes});
            this.$router.push({path: action.path});
        },
    },
});
