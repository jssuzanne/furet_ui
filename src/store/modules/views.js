/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
export const defaultState = {};

// getters
export const getters = {
};

// actions
export const actions = {
};

// mutations
export const mutations = {
    'UPDATE_VIEW'(state, action) {
        const value = Object.assign({}, action);
        delete value.viewId;
        const values = {};
        values[action.viewId] = Object.assign({}, state[action.viewId], value);
        Object.assign(state, values);
    },
    'CLEAR_VIEW'(state, action) {
        _.each(_.keys(state), k => delete state[k]);
        Object.assign(state, defaultState);
    },
};

export default {
  state: defaultState,
  getters,
  actions,
  mutations
}
