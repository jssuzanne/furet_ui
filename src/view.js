/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import Vue from 'vue';
import plugin from './plugin';
import uuid from 'uuid/v1';

export const getNewID = (model) => {
    return 'new-' + model + '-' + uuid();
}

/**
 * Return the stanadrd view
 *
 * @viewId: (string) id of the view
 * @params: object
 *
**/
export const View = Vue.component('furet-ui-view', {
    props: ['spaceId', 'menuId', 'actionId', 'viewId', 'dataId', 'mode'],
    render: function(createElement) {
        let view = plugin.get(['views', 'type', this.view.viewType]);
        if (!view) view = plugin.get(['views', 'Unknown']).vue;
        return createElement(view, {
            props: {
                spaceId: this.spaceId,
                menuId: this.menuId,
                actionId: this.actionId,
                viewId: this.viewId,
                view: this.view,
                viewName: this.view.viewType,
                dataId: this.dataId,
                data: this.data,
                change: this.change,
                mode: this.mode,
            },
        });
    },
    computed: {
        view () {
            const views = this.$store.state.data.views;
            if (this.viewId) return views[String(this.viewId)] || {};
            return {};
        },
        model () {
            if (this.view) {
                return this.view.model;
            }
            return null;
        },
        data () {
            if (this.model) {
                let data = this.$store.state.data.data;
                let changes = this.$store.state.data.changes;

                if (this.dataId) {
                    data = (data[this.model]) ? data[this.model][String(this.dataId)] || {} : {};
                    changes = (changes[this.model]) ? changes[this.model][String(this.dataId)] || {} : {};
                    return Object.assign({}, data, changes);
                }

                return data[this.model] || {};
            }
            if (this.dataId) return {};
            return [];
        },
        change () {
            if (this.dataId) {
                const changes = this.$store.state.data.changes;
                return (changes[this.model]) ? changes[this.model][String(this.dataId)] || {} : {};
            }
            return {};
        },
    },
});


/**
 * Return the Icon component which represent the standard view
 *
 * @type: typs of the standard view
 *
**/
export const ViewIcon = Vue.component('furet-ui-view-icon', {
    props: ['type'],
    render: function(createElement) {
        let view = plugin.get(['views', 'icon', this.type]);
        if (!view) view = plugin.get(['views', 'icon', 'Unknown']);
        return createElement(view, {});
    }
});

/**
 * Return a component with the wanted view, if no view the the return view will be the Unknown view
 *
 * @viewName: name of the wanted view
 *
**/
export const CustomView = Vue.component('furet-ui-custom-view', {
    props: ['viewName'],
    render: function(createElement) {
        let view = plugin.get(['views', 'type', 'client', this.viewName]);
        if (!view) view = plugin.get(['views', 'Unknown']);
        if (view.vue) return createElement(view.vue, {props: {viewName: this.viewName}});
        else if (view.function) return view.function({
            createElement, store: this.$store, router: this.$router});
    }
});
