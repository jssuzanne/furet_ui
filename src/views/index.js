/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
// import AlertWarning from 'material-ui/svg-icons/alert/warning';
// 
// /**
//  * Unknown icon for view selector
// **/
// plugin.set(['views', 'icon'], {Unknown: (props) => {
//     return <AlertWarning />;
// }});

import './clients';
// import './list';
// import './thumbnail';
// import './form';
import Vue from 'vue';
import plugin from '../plugin';

/**
 * Unknown view, use if no view found
 *
 * props:
 *  @selector: Component used by the view
 *  @viewName: Name of the custom view which is not available
 *  @viewType: Name of the standard view which is not available
 *
**/
                // <div className="col-xs-12 col-sm-8 col-md-8 col-lg-9">
                // </div>
                // <div className="col-xs-12 col-sm-4 col-md-4 col-lg-3">
                //     {this.props.selector || null}
                // </div>
export const VueUnknown = Vue.component('furet-ui-view-unknown', {
    props: ['selector', 'viewName'],
    template: `
        <div>
            <div v-if="selector">
                {{selector}}
            </div>
            <section class="section">
                <div class="container">
                    <h1 class="title">{{$t('views.unknown.title', {name: viewName})}}</h1>
                    <h2 class="subtitle">
                        {{$t('views.unknown.message')}}
                    </h2>
                </div>
            </section>
        </div>`,
});

plugin.set(['views'], {Unknown: {vue: 'furet-ui-view-unknown'}})
