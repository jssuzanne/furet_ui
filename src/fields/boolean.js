// /**
// This file is a part of the FuretUI project
// 
//    Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>
// 
// This Source Code Form is subject to the terms of the Mozilla Public License,
// v. 2.0. If a copy of the MPL was not distributed with this file,You can
// obtain one at http://mozilla.org/MPL/2.0/.
// **/
import Vue from 'vue';
import plugin from '../plugin';
import {FormMixin, ThumbnailMixin} from './common';
import {i18n} from '../i18n';

Vue.component('furet-ui-list-field-boolean', {
    props: ['checked'],
    template: '<b-checkbox v-bind:checked="checked" disabled></b-checkbox>',
})


plugin.set(['field', 'List'], {Boolean: (header) => {
    const res = {
        label: header.label,
        field: header.name,
        width: 40,
        renderHtml: (createElement, row) => {
            return createElement('furet-ui-list-field-boolean', {
                props: {checked: eval(row[header.name]) ? true : false}
            });
        },
    }
    if (header.sortable) res.sortable = header.sortable;
    return res;
}})

export const FieldThumbnailBoolean = Vue.component('furet-ui-thumbnail-field-boolean', {
    props: ['name', 'label', 'params', 'data'],
    mixins: [ThumbnailMixin],
    template: `
        <div v-if="this.isInvisible" />
        <b-checkbox 
            v-else
            v-bind:checked="value" 
            disabled
        >
            {{this.label}}
        </b-checkbox>`,
    computed: {
        value () {
            return eval(this.data[this.name]) ? true : false;
        },
    }
})
plugin.set(['field', 'Thumbnail'], {Boolean: 'furet-ui-thumbnail-field-boolean'});

export const FieldFormBoolean = Vue.component('furet-ui-form-field-boolean', {
    props: ['name', 'label', 'params', 'config'],
    mixins: [FormMixin],
    template: `
        <div v-if="this.isInvisible" />
        <b-checkbox 
            v-else
            v-bind:checked="data" 
            v-bind:disabled="isReadonly"
        >
            {{this.label}}
        </b-checkbox>`,
    computed: {
        data () {
            const value = this.config && this.config.data && this.config.data[this.name] || '';
            return eval(value) ? true : false;
        },
    },
})
plugin.set(['field', 'Form'], {Boolean: 'furet-ui-form-field-boolean'});
