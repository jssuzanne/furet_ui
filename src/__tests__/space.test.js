/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import Vue from 'vue';
import Buefy from 'buefy';
Vue.use(Buefy, {defaultIconPack: 'fa',});
import {store} from '../store';
import {router} from '../routes';
import {i18n} from '../i18n';
import '../view';
import '../views';
import '../fields';
import {ViewSelector, SpaceMenu, Space} from '../space';

jest.mock('../server-call')
const menus = [
    {
        label: 'Customer',
        image: {'type': 'font-icon', 'value': 'fa-user'},
        actionId: '2',
        id: '1',
        submenus: [],
    },
    {
        label: 'Setting',
        image: {'type': 'font-icon', 'value': 'fa-user'},
        actionId: '',
        id: '2',
        submenus: [
            {
                label: 'Category',
                image: {'type': '', 'value': ''},
                actionId: '3',
                id: '3',
                submenus: [],
            },
            {
                label: 'Address',
                image: {'type': '', 'value': ''},
                actionId: '4',
                id: '4',
                submenus: [],
            },
        ],
    },
]

describe('View Selector component', () => {
    const renderer = require('vue-server-renderer').createRenderer();
    beforeEach(() => {
        store.dispatch('UNITEST_CLEAR');
        router.push({path: '/'});
    });
    it('Render', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(Space, {params: {
                viewId: '1',
                views: [
                    {viewId: '1', type: 'List'},
                    {viewId: '2', type: 'Form', unclickable: true},
                ],
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render unclickable selected', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(Space, {params: {
                viewId: '2',
                views: [
                    {viewId: '1', type: 'List'},
                    {viewId: '2', type: 'Form', unclickable: true},
                ],
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
});

describe('SpaceMenu component', () => {
    const renderer = require('vue-server-renderer').createRenderer();
    beforeEach(() => {
        store.dispatch('UNITEST_CLEAR');
        router.push({path: '/'});
    });
    it('Render', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(SpaceMenu),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render with submenu', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(SpaceMenu, {props: {
                menuId: '1',
                menus,
                spaceId: '1',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render with submenu other selected menu', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(SpaceMenu, {props: {
                menuId: '3',
                menus,
                spaceId: '1',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
});

describe('Space component', () => {
    const renderer = require('vue-server-renderer').createRenderer();
    beforeEach(() => {
        store.dispatch('UNITEST_CLEAR');
        router.push({path: '/'});
    });
    it('Render App with default path', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(Space),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
});

// test('Render Space with default value from redux store with spaceId', () => {
//     const store = createStore(combineReducers(reducers));
//     updateGlobal();
//     const space = require('../space'),
//           Space = space.default;
//     const component = renderer.create(
//         <Provider store={store}>
//             <MuiThemeProvider>
//                 <Space spaceId='1'/>
//             </MuiThemeProvider>
//         </Provider>
//     );
//     store.dispatch({
//         'type': 'UPDATE_SPACE',
//         'spaceId': '1',
//         'menuId': '1',
//         'actionId': '1',
//         'viewId': '1',
//         'custom_view': '',
//     });
//     let tree = component.toJSON();
//     expect(tree).toMatchSnapshot();
// });
// 
// test('Render Space with default value from redux store with spaceId with menu', () => {
//     const store = createStore(combineReducers(reducers));
//     updateGlobal();
//     const space = require('../space'),
//           Space = space.default;
//     const component = shallow(
//         <Provider store={store}>
//             <MuiThemeProvider>
//                 <Space spaceId='1'/>
//             </MuiThemeProvider>
//         </Provider>
//     );
//     const getMenu = (label, spaceId) => {
//         return [
//             {
//                 'label': 'Menu ' + label + ' 1 : ' + spaceId,
//                 'image': {'type': 'font-icon', 'value': 'fa-user'},
//                 'actionId': '1',
//                 'custom_view': '',
//                 'id': label + '1',
//                 'submenus': [],
//             },
//             {
//                 'label': 'Menu ' + label + ' 2 : ' + spaceId,
//                 'image': {'type': '', 'value': ''},
//                 'actionId': '',
//                 'custom_view': 'Login',
//                 'id': label + '3',
//                 'submenus': [],
//             },
//             {
//                 'label': 'Menu ' + label + ' 2 : ' + spaceId,
//                 'image': {'type': '', 'value': ''},
//                 'actionId': '',
//                 'custom_view': '',
//                 'id': label + '4',
//                 'submenus': [
//                     {
//                         'label': 'Menu ' + label + ' 1 : ' + spaceId,
//                         'image': {'type': 'font-icon', 'value': 'fa-user'},
//                         'actionId': '2',
//                         'custom_view': '',
//                         'id': label + '41',
//                         'submenus': [],
//                     },
//                 ],
//             },
//         ]
//     }
//     store.dispatch({
//         'type': 'UPDATE_SPACE',
//         'spaceId': '1',
//         'left_menu': getMenu('left', '1'),
//         'menuId': '1',
//         'right_menu': getMenu('right', '1'),
//         'actionId': '1',
//         'viewId': '1',
//         'custom_view': '',
//     });
//     let tree = toJson(component);
//     expect(tree).toMatchSnapshot();
// });
// 
// test('Render Space with default value from redux store with custom client', () => {
//     const store = createStore(combineReducers(reducers));
//     updateGlobal();
//     const space = require('../space'),
//           Space = space.default;
//     const component = renderer.create(
//         <Provider store={store}>
//             <MuiThemeProvider>
//                 <Space spaceId='1'/>
//             </MuiThemeProvider>
//         </Provider>
//     );
//     store.dispatch({
//         'type': 'UPDATE_SPACE',
//         'spaceId': '',
//         'menuId': '1',
//         'actionId': '',
//         'viewId': '1',
//         'custom_view': 'Login',
//     });
//     let tree = component.toJSON();
//     expect(tree).toMatchSnapshot();
// });
