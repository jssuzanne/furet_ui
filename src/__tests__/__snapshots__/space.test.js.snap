// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Space component Render App with default path 1`] = `
<div data-server-rendered="true" class="columns is-gapless">
    <!---->
    <div class="column" style="padding-left:10px;padding-right:10px;">
        <nav class="nav">
            <div class="nav-left">
                <!---->
                <ul style="padding:10px 0px;list-style:none;">
                    <li style="display:inline;">

                    </li>
                </ul>
            </div>
            <div class="nav-right">
                <!---->
                <!---->
            </div>
        </nav>
        <!---->
    </div>
    <!---->
</div>
`;

exports[`SpaceMenu component Render 1`] = `<ul data-server-rendered="true" class="menu-list"></ul>`;

exports[`SpaceMenu component Render with submenu 1`] = `
<ul data-server-rendered="true" class="menu-list">
    <li>
        <!---->
        <!----><a class="is-active"><span class="icon"><i class="fa fa-user"></i></span>
                    Customer
                </a></li>
    <li><span><span class="icon"><i class="fa fa-user"></i></span> Setting
        </span>
        <ul class="menu-list">
            <li>
                <!---->
                <!---->
                <a>
                    <!---->
                    Category
                </a>
            </li>
            <li>
                <!---->
                <!---->
                <a>
                    <!---->
                    Address
                </a>
            </li>
        </ul>
        <!---->
    </li>
</ul>
`;

exports[`SpaceMenu component Render with submenu other selected menu 1`] = `
<ul data-server-rendered="true" class="menu-list">
    <li>
        <!---->
        <!----><a><span class="icon"><i class="fa fa-user"></i></span>
                    Customer
                </a></li>
    <li><span><span class="icon"><i class="fa fa-user"></i></span> Setting
        </span>
        <ul class="menu-list">
            <li>
                <!---->
                <!---->
                <a class="is-active">
                    <!---->
                    Category
                </a>
            </li>
            <li>
                <!---->
                <!---->
                <a>
                    <!---->
                    Address
                </a>
            </li>
        </ul>
        <!---->
    </li>
</ul>
`;

exports[`View Selector component Render 1`] = `
<div data-server-rendered="true" class="columns is-gapless">
    <!---->
    <div class="column" style="padding-left:10px;padding-right:10px;">
        <nav class="nav">
            <div class="nav-left">
                <!---->
                <ul style="padding:10px 0px;list-style:none;">
                    <li style="display:inline;">

                    </li>
                </ul>
            </div>
            <div class="nav-right">
                <!---->
                <!---->
            </div>
        </nav>
        <!---->
    </div>
    <!---->
</div>
`;

exports[`View Selector component Render unclickable selected 1`] = `
<div data-server-rendered="true" class="columns is-gapless">
    <!---->
    <div class="column" style="padding-left:10px;padding-right:10px;">
        <nav class="nav">
            <div class="nav-left">
                <!---->
                <ul style="padding:10px 0px;list-style:none;">
                    <li style="display:inline;">

                    </li>
                </ul>
            </div>
            <div class="nav-right">
                <!---->
                <!---->
            </div>
        </nav>
        <!---->
    </div>
    <!---->
</div>
`;
